doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -1462,64 +1375,65 @@ Abbreviations
 Numeral notations
 -----------------
 
-.. cmd:: Numeral Notation @ident__1 @ident__2 @ident__3 : @scope.
+.. cmd:: Numeral Notation @qualid @qualid__parse @qualid__print : @scope {? @numnotoption }
    :name: Numeral Notation
 
+   .. insertprodn numnotoption numnotoption
+
+   .. prodn::
+      numnotoption ::= ( warning after @numeral )
+      | ( abstract after @numeral )
+
    This command allows the user to customize the way numeral literals
    are parsed and printed.
 
-   The token :n:`@ident__1` should be the name of an inductive type,
-   while :n:`@ident__2` and :n:`@ident__3` should be the names of the
+   :n:`@qualid` should be the name of an inductive type,
+   while :n:`@qualid__parse` and :n:`@qualid__print` should be the names of the
    parsing and printing functions, respectively.  The parsing function
-   :n:`@ident__2` should have one of the following types:
-
-     * :n:`Decimal.int -> @ident__1`
-     * :n:`Decimal.int -> option @ident__1`
-     * :n:`Decimal.uint -> @ident__1`
-     * :n:`Decimal.uint -> option @ident__1`
-     * :n:`Z -> @ident__1`
-     * :n:`Z -> option @ident__1`
-     * :n:`Decimal.decimal -> @ident__1`
-     * :n:`Decimal.decimal -> option @ident__1`
-
-   And the printing function :n:`@ident__3` should have one of the
+   :n:`@qualid__parse` should have one of the following types:
+
+     * :n:`Decimal.int -> @qualid`
+     * :n:`Decimal.int -> option @qualid`
+     * :n:`Decimal.uint -> @qualid`
+     * :n:`Decimal.uint -> option @qualid`
+     * :n:`Z -> @qualid`
+     * :n:`Z -> option @qualid`
+     * :n:`Decimal.decimal -> @qualid`
+     * :n:`Decimal.decimal -> option @qualid`
+
+   And the printing function :n:`@qualid__print` should have one of the
    following types:
 
-     * :n:`@ident__1 -> Decimal.int`
-     * :n:`@ident__1 -> option Decimal.int`
-     * :n:`@ident__1 -> Decimal.uint`
-     * :n:`@ident__1 -> option Decimal.uint`
-     * :n:`@ident__1 -> Z`
-     * :n:`@ident__1 -> option Z`
-     * :n:`@ident__1 -> Decimal.decimal`
-     * :n:`@ident__1 -> option Decimal.decimal`
+     * :n:`@qualid -> Decimal.int`
+     * :n:`@qualid -> option Decimal.int`
+     * :n:`@qualid -> Decimal.uint`
+     * :n:`@qualid -> option Decimal.uint`
+     * :n:`@qualid -> Z`
+     * :n:`@qualid -> option Z`
+     * :n:`@qualid -> Decimal.decimal`
+     * :n:`@qualid -> option Decimal.decimal`
 
      When parsing, the application of the parsing function
-     :n:`@ident__2` to the number will be fully reduced, and universes
+     :n:`@qualid__parse` to the number will be fully reduced, and universes
      of the resulting term will be refreshed.
 
      Note that only fully-reduced ground terms (terms containing only
      function application, constructors, inductive type families,
      sorts, and primitive integers) will be considered for printing.
```

Comment:
Fixed.  BTW, we seem to use 3 space indentation a lot (e.g. for directives).  Should we only use 3 space increments in the rsts?  It's a bit odd to use 3.  2 is easier to deal with.

Reply:
3 spaces is the standard usually, but it's not mandated.  See https://stackoverflow.com/questions/48311159/is-3-space-indentation-required-in-rest/48313531

Reply:
3 spaces doesn't work so well for me.  In my IDE I can set the indentation independently for a short list of predefined file types, including Python, then everything else is lumped under "Other File Types".  So setting to 3 would also apply to .ml files, which would not be good.  It's especially awkward when I want to change the indentation of a paragraph--I end up doing it line by line.

Reply:
I think that only indenting two spaces is ugly because it is badly aligned with the documented object (the three spaces correspond to `..` + a space).
I suggest that you use another IDE that supports the rst format for doing your edits to the manual. Having syntax highlighting would avoid you to do mistakes like forgetting a colon or backtick. Emacs works well for me (without any extension).

Reply:
> I suggest that you use another IDE
> Emacs works well for me (without any extension).

Not gonna happen.  Definitely not a fan of emacs.  A topic that raises strong passions :-)

How about this in the doc for the Arguments command?  It mixes 3- and 2- indenting.

```
   ..

     `/`
       the function will be unfolded only if it's applied to at least the
       arguments appearing before the `/`.  See :ref:`Args_effect_on_unfolding`.
```

Reply:
What is the `..` for BTW? It's not needed to create such definition lists.

Anyway, I am aware that the documentation is not consistent everywhere regarding indentation, but that seems a minor thing to me (but I would prefer to be consistent whenever we introduce new indented code).

Note that I am of course not forcing you to change your IDE, and certainly not to use Emacs, but the fact that you are using an IDE which has poor support for Sphinx should not be an argument for deviating from the standard practices (which are well supported in other IDEs).

Reply:
Well, I would not have asked about indentation at all if I didn't intend to do the right thing.  I take it then that we prefer consistent indent of 3 for new and changed material.

> What is the `..` for BTW?

Some Sphinx doc suggested it was needed to get the definition list style.  Apparently not needed or I misunderstood.

Reply:
For bullet lists, 3-spacing requires an extra space after the `*`:

```
   *  Making arguments implicit. Afterward, implicit arguments
      must be omitted in any expression that applies :token:`smart_qualid`.
```

If there's only one space, the output is different:

![image](https://user-images.githubusercontent.com/1253341/80315321-9bb96f80-87ab-11ea-836b-ef1cefddaba9.png)

compared to:

![image](https://user-images.githubusercontent.com/1253341/80315346-c4da0000-87ab-11ea-888a-8d8e26be17a2.png)

Reply:
The rule should not always be "three spaces" but always as many spaces as necessary to align on the first line. So in the case of bullet-lists, two spaces is good. @cpitclaudel has more experience than me on this format and should confirm this.

Reply:
How about this?  OK or should be 3 spaces after the first line?  Perhaps better not to have text on the same line as `.. note::`?

```
   .. note:: When you are referring to hypotheses which you did not name
             explicitly, be aware that Coq may make a different decision on how to
             name the variable in the current goal and in the context of the
             existential variable. This can lead to surprising behaviors.
```

And this should be 3 spaced after the first line, right?

```
      :n:`@name {? % @scope }`
        a *formal parameter* of the function :n:`@smart_qualid` (i.e.
        the parameter name used in the function definition).  Unless `rename` is specified,
        the list of :n:`@name`\s must be a prefix of the formal parameters, including all implicit
        arguments.  `_` can be used to skip over a formal parameter.
        :token:`scope` can be either a scope name or its delimiting key.  See :ref:`binding_to_scope`.

```